src/Uno.UI/Microsoft/UI/Xaml/Controls/RatingControl/RatingControl.cs

```diff
@@ -0,0 +1,1195 @@
+
+				double mousePercentage = xPosition / CalculateActualRatingWidth();
+				SetRatingTo(Math.Ceiling(mousePercentage * MaxRating), true);
+
+				if (SharedHelpers.IsRS1OrHigher())
+				{
+					ElementSoundPlayer.Play(ElementSoundKind.Invoke);
+				}
+			}
+
+			if (m_isPointerDown)
+			{
+				m_isPointerDown = false;
+				UpdateRatingItemsAppearance();
+			}
+		}
+
+		private double CalculateTotalRatingControlWidth()
+		{
+			double ratingStarsWidth = CalculateActualRatingWidth();
+			var captionAsWinRT = (string)GetValue(CaptionProperty);
+			double textSpacing = 0.0;
+
+			if (captionAsWinRT?.Length > 0)
+			{
+				textSpacing = ItemSpacing();
+			}
+
+			double captionWidth = 0.0;
+
+			if (m_captionTextBlock != null)
+			{
+				captionWidth = m_captionTextBlock.ActualWidth;
+			}
+
+			return ratingStarsWidth + textSpacing + captionWidth;
+		}
+
+		double CalculateStarCenter(int starIndex)
+		{
+			// TODO: sub in real API DP values
+			// MSFT #10030063
+			// [real Rating Size * (starIndex + 0.5)] + (starIndex * itemSpacing)
+			return (ActualRatingFontSize() * (starIndex + 0.5)) + (starIndex * ItemSpacing());
+		}
+
+		private double CalculateActualRatingWidth()
+		{
+			// TODO: replace hardcoding
+			// MSFT #10030063
+			// (max rating * rating size) + ((max rating - 1) * item spacing)
+			return (MaxRating * ActualRatingFontSize()) + ((MaxRating - 1) * ItemSpacing());
+		}
+
+		// IControlOverrides
+		protected override void OnKeyDown(KeyRoutedEventArgs eventArgs)
+		{
+			if (eventArgs.Handled)
+			{
+				return;
+			}
+
+			if (!IsReadOnly)
+			{
+				bool handled = false;
+				VirtualKey key = eventArgs.Key;
+
+				double flowDirectionReverser = 1.0;
+
+				if (FlowDirection == FlowDirection.RightToLeft)
+				{
+					flowDirectionReverser *= -1.0;
+				}
+
+				var originalKey = eventArgs.OriginalKey;
+
+				// Up down are right/left in keyboard only
+				if (originalKey == VirtualKey.Up)
+				{
+					key = VirtualKey.Right;
+					flowDirectionReverser = 1.0;
+				}
+				else if (originalKey == VirtualKey.Down)
+				{
+					key = VirtualKey.Left;
+					flowDirectionReverser = 1.0;
+				}
+
+				if (originalKey == VirtualKey.GamepadDPadLeft || originalKey == VirtualKey.GamepadDPadRight ||
+					originalKey == VirtualKey.GamepadLeftThumbstickLeft || originalKey == VirtualKey.GamepadLeftThumbstickRight)
+				{
+					if (SharedHelpers.IsRS1OrHigher())
+					{
+						ElementSoundPlayer.Play(ElementSoundKind.Focus);
+					}
+				}
+
+				switch (key)
+				{
+					case VirtualKey.Left:
+						ChangeRatingBy(-1.0 * flowDirectionReverser, false);
+						handled = true;
+						break;
+					case VirtualKey.Right:
+						ChangeRatingBy(1.0 * flowDirectionReverser, false);
+						handled = true;
+						break;
+					case VirtualKey.Home:
+						SetRatingTo(0.0, false);
+						handled = true;
+						break;
+					case VirtualKey.End:
+						SetRatingTo((double)(MaxRating), false);
+						handled = true;
+						break;
+					default:
```

Comment:
Codacy found an issue: [Remove this empty 'default' clause.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)